src/offlineActionTracker.js

```diff
@@ -1,24 +1,24 @@
-const subscriptions = [];
+const subscriptions = {};
 
 function registerAction(transaction) {
   return new Promise((resolve, reject) => {
-    subscriptions.push({ transaction, resolve, reject });
+    subscriptions[transaction] = { resolve, reject };
   });
 }
 
 function resolveAction(transaction, value) {
-  const subscription = subscriptions[0];
-  if (subscription && subscription.transaction === transaction) {
+  const subscription = subscriptions[transaction];
+  if (subscription) {
     subscription.resolve(value);
-    subscriptions.shift();
+    delete subscriptions[transaction];
```

Comment:
In general I prefer to avoid using `delete` as it's a slow operation that v8 etc struggles to optimise. 
You might be better to destructure and reassign subscriptions to remove properties from the subscriptions object.

Reply:
If we did that, subscriptions would have to be a `let` and not a `const`, right?

Reply:
Correct - most likely a bit premature at this point however, feel free to ignore. 

(might be a case for `new Map()` in the future as it is a dictionary, just a thought)